### PR TITLE
Conform to XMLHttpRequest spec

### PR DESCRIPTION
**Pull request by jiangts**
According to [the spec](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest), `XMLHttpRequest.getAllResponseHeaders()` and `XMLHttpRequest.getResponseHeader()` should return `null` if the header does not exist.

The current implementation returns `undefined`, which breaks code that assumes the correct implementation of the spec, such as [the check for null in this library function](https://github.com/mholt/PapaParse/blob/master/papaparse.js#L643).

### CODE REVIEW

**Comment by jpillora** [comment]
Hey @morsdyce do you know anything about the addition of the `window` object to the IIFE?

**Comment by morsdyce** [comment]
Yes, I remove it manually before every release since cofffescript compiler
forces it and there's no way to remove it automatically.

Will be changed once the transition to es6 will be completed

**Comment by jiangts** [comment]
curious -- what problem does the addition of `window` cause?

or is the main problem that it assumes a browser execution context

**Comment by morsdyce** [comment]
It assumes browser context where xhook can run on different platforms such
as web workers and react native

File: dist/xhook.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 // XHook - v1.4.4 - https://github.com/jpillora/xhook
 // Jaime Pillora <dev@jpillora.com> - MIT Copyright 2017
-(function(undefined) {
-var AFTER, BEFORE, COMMON_EVENTS, EventEmitter, FETCH, FIRE, FormData, NativeFetch, NativeFormData, NativeXMLHttp, OFF, ON, READY_STATE, UPLOAD_EVENTS, WINDOW, XHookFetchRequest, XHookFormData, XHookHttpRequest, XMLHTTP, convertHeaders, depricatedProp, document, fakeEvent, mergeObjects, msie, proxyEvents, slice, useragent, xhook, _base,
+(function(window,undefined) {
```
Comment (jpillora) [inline dist/xhook.js]: Could you please manually delete `window` from here and from below?

Comment (jiangts) [inline dist/xhook.js]: done, and in the minified build too

**Comment by jpillora** [comment]
Thanks :)

**Pull request merged**